source/game/system/map/MapdataCannonPoint.hh

```diff
@@ -0,0 +1,88 @@
+// lol these names x.x vabold help me
+
+#pragma once
+
+#include "game/system/map/MapdataAccessorBase.hh"
+
+#include <egg/math/Vector.hh>
+
+namespace System {
+class MapdataCannonPoint {
+public:
+    struct CannonParameter {
+        f32 speed;
+        f32 height;
+        f32 decelFactor;
+        f32 endDecel;
+    };
+
+    struct SData {
+        EGG::Vector3f pos;
+        EGG::Vector3f rot;
+        u16 id;
+        s16 propertyIdx;
+    };
+
+    MapdataCannonPoint(const SData *data);
+    void read(EGG::Stream &stream);
+
+    /// @beginGetters
+    const EGG::Vector3f pos() const {
+        return m_pos;
+    }
+    const EGG::Vector3f rot() const {
+        return m_rot;
+    }
+    u16 id() const {
+        return m_id;
+    }
+    const CannonParameter parameters() const {
+        assert(m_paramterIdx <= 2);
+        // i don't know why negative cannon indeces are treated as 0, but they are.
```

Comment:
Transform this into a `/// @brief` comment. Also none of the functions here should have their bodies in the header. Move them to the source please